skimage/util/montage.py

```diff
@@ -78,11 +81,16 @@ def montage2d(arr_in, fill='mean', rescale_intensity=False, grid_shape=None):
     """
 
     assert arr_in.ndim == 3
-
+    
+    # -- add border padding, np.pad does all dimensions 
+    # so we remove the padding from the first
+    if border_padding > 0:
+        arr_in = np.pad(arr_in, border_padding, mode='constant')[border_padding:-border_padding]
```

Comment:
As mentioned by @soupault , ``pad_width`` in ``np.pad`` can be an iterable, so you can specify 0-padding along the 1st dimension inside ``np.pad``.